packages/block-renderer/src/components/block-renderer-container.tsx

```diff
@@ -85,13 +89,30 @@ const ScaledBlockRendererContainer = ( {
 
 	const scale = containerWidth / viewportWidth;
 
+	let scaledHeight = ( contentHeight as number ) * scale || minHeight;
+	if ( isMinHeight100vh && maxHeightFor100vh && ! viewportHeight ) {
+		scaledHeight = maxHeightFor100vh * scale;
+	}
+
+	let iframeHeight = ( contentHeight as number ) || minHeight;
+	if ( isMinHeight100vh ) {
+		if ( viewportHeight ) {
+			iframeHeight = viewportHeight;
+		} else if ( maxHeightFor100vh ) {
+			iframeHeight = maxHeightFor100vh;
+		}
+	}
+
 	return (
 		<div
 			className="scaled-block-renderer"
 			style={ {
 				transform: `scale(${ scale })`,
-				height: ( contentHeight as number ) * scale || minHeight,
-				maxHeight: ( contentHeight as number ) > maxHeight ? maxHeight * scale : undefined,
+				height: scaledHeight,
+				maxHeight:
+					maxHeight !== 'none' && ( contentHeight as number ) > maxHeight
+						? ( maxHeight as number ) * scale
+						: undefined,
```

Comment:
Hey @miksansegundo, I noticed this while working on #74540. What are the units for maxHeight here? One issue I noticed is that `maxHeight` can technically be a string (e.g. "200px"), which probably wouldn't work with this logic. (But it doesn't seem like we use it like that anywhere.)

But the main issue is that the types are technically wrong, since we now allow `maxHeight` to be a _string_ or a number. If we only want to allow "none", could we change it so that the prop type is `"none" | number`?

Reply:
Thanks for the ping @noahtallen, the units for `maxHeight ` are `pixels` and the type is `number`. I added the type `string` only to support disabling the default value using `none` . 

I'll change the type to use `'none' | number` for now and follow up with a better way to disable the default value via a new prop. See https://github.com/Automattic/wp-calypso/pull/74843

Reply:
Cool!